File: VB/ColoredProgressBar/SingleRepositoryProgressBar.vb
Imports DevExpress.XtraEditors
Imports DevExpress.XtraEditors.ViewInfo
Imports DevExpress.XtraGrid.Columns
Imports DevExpress.XtraGrid.Views.Grid
Imports DevExpress.XtraGrid.Views.Grid.ViewInfo
Imports System
Imports System.Drawing

Namespace ColoredProgressBar

    Friend Class SRProgressBarHelper

        Private _Column As GridColumn

        Private _View As GridView

        Public Sub New(ByVal column As GridColumn)
            _Column = column
            _View = TryCast(_Column.View, GridView)
            AddHandler _View.ShownEditor, New EventHandler(AddressOf View_ShownEditor)
            AddHandler _View.CustomDrawCell, New DevExpress.XtraGrid.Views.Base.RowCellCustomDrawEventHandler(AddressOf View_CustomDrawCell)
        End Sub

        Private Sub View_ShownEditor(ByVal sender As Object, ByVal e As EventArgs)
            If _View.FocusedColumn IsNot _Column Then Return
            Dim pbc As ProgressBarControl = TryCast(_View.ActiveEditor, ProgressBarControl)
            Dim percent As Integer = Convert.ToInt16(pbc.EditValue)
            If percent < 25 Then
                pbc.Properties.EndColor = Color.Red
                pbc.Properties.StartColor = Color.Red
            ElseIf percent < 50 Then
                pbc.Properties.EndColor = Color.Orange
                pbc.Properties.StartColor = Color.Orange
            ElseIf percent < 75 Then
                pbc.Properties.EndColor = Color.YellowGreen
                pbc.Properties.StartColor = Color.YellowGreen
            ElseIf percent <= 100 Then
                pbc.Properties.EndColor = Color.Green
                pbc.Properties.StartColor = Color.Green
            End If
        End Sub

        Private Sub View_CustomDrawCell(ByVal sender As Object, ByVal e As DevExpress.XtraGrid.Views.Base.RowCellCustomDrawEventArgs)
            If e.Column Is _Column Then
                Dim percent As Integer = Convert.ToInt16(e.CellValue)
                Dim vi As ProgressBarViewInfo = TryCast(TryCast(e.Cell, GridCellInfo).ViewInfo, ProgressBarViewInfo)
                If percent < 25 Then
                    vi.ProgressInfo.EndColor = Color.Red
                    vi.ProgressInfo.StartColor = Color.Red
                ElseIf percent < 50 Then
                    vi.ProgressInfo.EndColor = Color.Orange
                    vi.ProgressInfo.StartColor = Color.Orange
                ElseIf percent < 75 Then
                    vi.ProgressInfo.EndColor = Color.YellowGreen
                    vi.ProgressInfo.StartColor = Color.YellowGreen
                ElseIf percent <= 100 Then
                    vi.ProgressInfo.EndColor = Color.Green
                    vi.ProgressInfo.StartColor = Color.Green
                End If
            End If
        End Sub
    End Class
End Namespace

File: CS/ColoredProgressBar/SingleRepositoryProgressBar.cs
using DevExpress.XtraEditors;
using DevExpress.XtraEditors.ViewInfo;
using DevExpress.XtraGrid.Columns;
using DevExpress.XtraGrid.Views.Grid;
using DevExpress.XtraGrid.Views.Grid.ViewInfo;
using System;
using System.Drawing;

namespace ColoredProgressBar {
    class SRProgressBarHelper {
        GridColumn _Column;
        GridView _View;
        public SRProgressBarHelper(GridColumn column) {
            _Column = column;
            _View = _Column.View as GridView;
            _View.ShownEditor += new EventHandler(View_ShownEditor);
            _View.CustomDrawCell += new DevExpress.XtraGrid.Views.Base.RowCellCustomDrawEventHandler(View_CustomDrawCell);
        }
        private void View_ShownEditor(object sender, EventArgs e) {
            if (_View.FocusedColumn != _Column)
                return;
            ProgressBarControl pbc = _View.ActiveEditor as ProgressBarControl;
            int percent = Convert.ToInt16(pbc.EditValue);
            if (percent < 25) {
                pbc.Properties.EndColor = Color.Red;
                pbc.Properties.StartColor = Color.Red;
            } else if (percent < 50) {
                pbc.Properties.EndColor = Color.Orange;
                pbc.Properties.StartColor = Color.Orange;
            } else if (percent < 75) {
                pbc.Properties.EndColor = Color.YellowGreen;
                pbc.Properties.StartColor = Color.YellowGreen;
            } else if (percent <= 100) {
                pbc.Properties.EndColor = Color.Green;
                pbc.Properties.StartColor = Color.Green;
            }
        }
        private void View_CustomDrawCell(object sender, DevExpress.XtraGrid.Views.Base.RowCellCustomDrawEventArgs e) {
            if (e.Column == _Column) {
                int percent = Convert.ToInt16(e.CellValue);
                ProgressBarViewInfo vi = (e.Cell as GridCellInfo).ViewInfo as ProgressBarViewInfo;
                if (percent < 25) {
                    vi.ProgressInfo.EndColor = Color.Red;
                    vi.ProgressInfo.StartColor = Color.Red;
                } else if (percent < 50) {
                    vi.ProgressInfo.EndColor = Color.Orange;
                    vi.ProgressInfo.StartColor = Color.Orange;
                } else if (percent < 75) {
                    vi.ProgressInfo.EndColor = Color.YellowGreen;
                    vi.ProgressInfo.StartColor = Color.YellowGreen;
                } else if (percent <= 100) {
                    vi.ProgressInfo.EndColor = Color.Green;
                    vi.ProgressInfo.StartColor = Color.Green;
                }
            }
        }
    }
}
